yarn-project/noir-protocol-circuits-types/src/index.ts

```diff
@@ -776,28 +776,33 @@ export function foreignCallHandler(name: string, args: ForeignCallInput[]): Prom
     const fieldsFr: Fr[] = fields.map((field: string) => fromACVMField(field));
     log.verbose('debug_log ' + applyStringFormatting(msg, fieldsFr));
   } else if (name === 'evaluateBlobs') {
-    // TODO(BLOB_ORACLE): this was added to save simulation time (~1min in ACVM, ~3mins in wasm -> 500ms).
+    // TODO(#10323): this was added to save simulation time (~1min in ACVM, ~3mins in wasm -> 500ms).
     // The use of bignum adds a lot of unconstrained code which overloads limits when simulating.
     // If/when simulation times of unconstrained are improved, remove this.
     // Create and evaulate our blobs:
-    const blobsAsFr: Fr[] = args[0].map((field: string) => fromACVMField(field)).filter(field => !field.isZero());
-    const blobs = Blob.getBlobs(blobsAsFr);
-    const blobPublicInputs = BlockBlobPublicInputs.fromBlobs(blobs);
-    // Checks on injected values:
+    const paddedBlobsAsFr: Fr[] = args[0].map((field: string) => fromACVMField(field));
     const kzgCommitments = args[1].map((field: string) => fromACVMField(field));
     const spongeBlob = SpongeBlob.fromFields(
       args
         .slice(2)
         .flat()
         .map((field: string) => fromACVMField(field)),
     );
+    const blobsAsFr = paddedBlobsAsFr.slice(0, spongeBlob.expectedFields);
+    // NB: the above used to be:
+    // const blobsAsFr: Fr[] = args[0].map((field: string) => fromACVMField(field)).filter(field => !field.isZero());
+    // ...but we now have private logs which have a fixed number of fields and may have 0 values.
+    // TODO(Miranda): trim 0 fields from private logs
```

Comment:
Now we have private logs which are 18 fields each, when the log doesn't use all 18 fields we introduce 0s in the blob. The above works for now but I will trim 0s and add a nested length prefix for each log once compile times go back down a bit.